cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetUpgrader.java

```diff
@@ -151,8 +153,13 @@ private boolean isCDAPUserTable(HTableDescriptor desc) {
     return defaultNSUserTablePrefix.matcher(tableName).matches() ||
       // Note: if the user has created a dataset called system.* then we will not upgrade the table.
       // CDAP-2977 should be fixed to have a cleaner fix for this.
-      !(tableName.contains("system.queue") || tableName.contains("system.stream") ||
-        tableName.contains("system.sharded.queue"));
+      !(isStreamOrQueueTable(tableName));
+  }
+
+  private boolean isStreamOrQueueTable(String tableName) {
```

Comment:
can we consolidate this with the code in DatasetAdminService#isUserDataset? Maybe into some util class?